### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by duncan-clark**
testing framework is now the same as lolog using testthat
devtools::test() runs all the tests nicely
also updated readme to describe a workflow for adding statistiscs

### CODE REVIEW

File: tests/testthat/test-stats.R [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    
+    net = make_net()
+    nets = lapply(1:100,FUN = make_net)
+    
+    
+    # ================
+    # Hamming Distance
+    # ================
+    
+    edges = as.edgelist(net)*1
```
Comment (ifellows) [inline tests/testthat/test-stats.R:42]: Not a required fix, but for good code style always use "<-" not "=".

File: tests/temp_adhoc_terms_testt.R [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #   set.vertex.attribute(net,"var_2",as.character(apply(rmultinom(100,1,rep(1/2,2)),2,FUN = function(x){which(x==1)})))
```
Comment (ifellows) [inline tests/temp_adhoc_terms_testt.R:22]: Is this file needed anymore or have these test scenarios been moved to the tests folder?

File: inst/include/Model.h [hunk 3]
```diff
@@ -514,21 +514,60 @@ class Model : public ShallowCopyable{
...
+    }
 
 	void discreteVertexUpdate(int vertex, int variable, int newValue){
 		for(int k=0;k<stats.size();k++)
 			stats[k]->vDiscreteVertexUpdate(*net,vertex, variable, newValue);
 		for(int k=0;k<offsets.size();k++)
 			offsets[k]->vDiscreteVertexUpdate(*net,vertex, variable, newValue);
 	}
+    
+    void discreteVertexUpdateR(int vertex, std::string varName, int newValue){
```
Comment (ifellows) [inline inst/include/Model.h:533]: I thought these changes were already merged in.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,15 @@
 # ernm
 Estimation of fully/partially observed Exponential-Family Random     Network Models (ERNM). ERNMs are a generalization of ERGM (see the ergm     package) and Gibbs Fields, encompassing both as special cases
+
+
+## Adding Statistics
```
Comment (ifellows) [inline README.md:5]: Note: The ideal way for a third party to add statistics is for them to create their own empty package, have it link to ernm, write just their stats into the package, and then register the stat on load by calling `register*Statistic`.

File: tests/temp_adhoc_terms_testt.R (hunk 2 above)
Comment (duncan-clark) [reply tests/temp_adhoc_terms_testt.R:22]: will remove, I will be adding these tests in the testthat style back in as I add more stats

File: README.md (hunk 4 above)
Comment (duncan-clark) [reply README.md:5]: I know this is the intended way - and makes sense for "specialized" stats. I was thinking to encourage users to make their stats available to all ERNM users? I'll add some words to this effect

File: inst/include/Model.h (hunk 3 above)
Comment (ifellows) [reply inst/include/Model.h:533]: I don't know. It is possible that it would look better if you were doing the pull requests from branches rather than a forked repo.

**Comment by ifellows** [comment]
looks good. Go ahead and merge when ready.
